test/cpp/runner/CppRunnerTests.cpp

```diff
@@ -1,12 +1,84 @@
 #include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
 #include "../../../src/cpp/runner/ModuleLoader.h"
+#include "../../../src/cpp/runner/Benchmark.h"
+#include "MockGMM.h"
+#include <thread>
+
+using ::chrono::seconds;
+using ::testing::_;
 
-TEST(CppRunnerTests, LibraryLoadTest) {
 #ifdef _DEBUG
-    ModuleLoader moduleLoader("MockGMMd.dll");
+const auto module_path = "MockGMMd.dll";
 #else
-    ModuleLoader moduleLoader("MockGMM.dll");
+const auto module_path = "MockGMM.dll";
 #endif
-	auto test = moduleLoader.get_gmm_test();
-    ASSERT_TRUE(test != NULL);
+
+ModuleLoader module_loader(module_path);
+
+TEST(CppRunnerTests, LibraryLoadTest) {
+   
+    ASSERT_TRUE(module_loader.get_gmm_test() != nullptr);
+}
+
+TEST(CppRunnerTests, TimeLimit) {
+
+    auto gmm_test = module_loader.get_gmm_test();
+    const auto minimum_measurable_time = 0s;
+    const auto run_count = 100; //Run count guarantees total time greater than the time_limit
+    const auto time_limit = 0.1s;
+    const auto execution_time = 0.01s;
+
```

Comment:
Please, add a comment about `execution_time` having to be greater than `minimum_measurable_time` for test assertions to be correct. (we need this so that `find_repeats...` won't do extra calls)